modules/editor/webhook/WebhookControls.tsx

```diff
@@ -68,6 +61,12 @@ export function WebhookControls(props: WebhookControlsProps) {
     }
   }, [])
 
+  const saveLabel = () => {
+    if (editorManager.messages.every(m => !m.reference)) return "Send"
+    if (editorManager.messages.every(m => m.reference)) return "Edit"
+    return "Submit"
+  }
```

Comment:
Don't use a function for this, instead a variable declared with `let` and mutate based it. When you do don't forget to use `else if` or you'll be looping through messages twice